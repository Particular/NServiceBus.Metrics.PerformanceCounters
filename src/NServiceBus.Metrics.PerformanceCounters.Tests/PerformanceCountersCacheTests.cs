namespace Tests
{
    using NServiceBus;
    using NUnit.Framework;

    [TestFixture]
    public class PerformanceCountersCacheTests
    {
        [Test]
        public void Get_the_same_counter_twice_returns_the_same()
        {
            var cache = new TestablePerformanceCountersCache();

            var counterName = "CounterName";
            var instanceName = "InstancenName";

            var firstCounter = cache.Get(new CounterInstanceName(counterName, instanceName));
            var secondCounter = cache.Get(new CounterInstanceName(counterName, instanceName));

            Assert.NotNull(firstCounter);
            Assert.NotNull(secondCounter);
            Assert.AreSame(firstCounter, secondCounter);
            Assert.AreEqual(1, cache.CountersCreated);
        }

        class TestablePerformanceCountersCache : PerformanceCountersCache
        {
            public int CountersCreated { get; private set; }

            protected override IPerformanceCounterInstance CreateInstance(CounterInstanceName counterInstanceName)
            {
                CountersCreated++;
                return new MockIPerformanceCounter();
            }
        }
    }
}