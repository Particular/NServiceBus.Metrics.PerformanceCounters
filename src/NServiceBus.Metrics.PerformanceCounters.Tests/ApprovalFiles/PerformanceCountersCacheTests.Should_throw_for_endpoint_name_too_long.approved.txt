The endpoint name ('*********************************************************************************************************************************') is too long (longer then 127) to register as a performance counter instance name. Shorten the endpoint name length.